### DISCUSSION TITLE
Trailing slashes on an endpoint?

### DISCUSSION THREAD

**Discussion by ghost** [counterfact/api-simulator#702] [Q&A]
Some of our codebase has endpoint references written with a trailing slash such as: `${myApiUrl}/someEndPoint/`. This throws off Counterfact because even though I have a file at `paths/someEndPoint.ts` that handles the need, Counterfact is expecting a subpath after the trailing slash. If I update the codebase to remove the trailing slash it works, but I'm trying to get my Counterfact setup to be a drop-in for the API that doesn't require the UX side to make changes just to accommodate Counterfact. Ideas?

**Comment by ghost** [comment]
Not sure this is the *right* fix, but it does fix it. In `module-tree.ts` if I change this:

```
    if (remainingSegments.length === 0) {
      return this.buildMatch(directory, segment, pathVariables, matchedPath);
    }
```
to this:
```
    if (remainingSegments.length === 0 || (remainingSegments.length === 1 && remainingSegments[0] === "")) {
      return this.buildMatch(directory, segment, pathVariables, matchedPath);
    }
```
it works. Basically the logic there is if my remaining segment is just any empty string then the app sent me a path with a trailing slash and no more path segments.

**Reply by pmcelhaney** [reply]
Yeah that makes sense. I can't think of a reason why presence or absence of trailing slash should make a difference.

**Reply by ghost** [reply]
Ok, PR is up: https://github.com/pmcelhaney/counterfact/pull/704
